immuneML/encodings/pdbencoding/PDBEncoder.py

```diff
@@ -0,0 +1,250 @@
+                    antigenID = i
+
+                collection[i].pop(0)
+
+
+            if lightID >= 0 and heavyID >= 0 and antigenID == -1:
+                if lightID + heavyID == 1:
+                    antigenID = 2
+
+                elif lightID + heavyID == 3:
+                    antigenID = 0
+
+            elif lightID == -1  or heavyID == -1 or antigenID == -1:
+                lightID = 0
+                heavyID = 1
+                antigenID = 2
+
+
+            lightChainDistanceToAntigen = np.zeros(shape=(len(collection[antigenID]), len(collection[lightID])))
+
+            caluculateDistance(collection, lightID, antigenID, lightChainDistanceToAntigen)
+
+            heavyChainDistanceToAntigen = np.zeros(shape=(len(collection[antigenID]), len(collection[heavyID])))
+            caluculateDistance(collection, heavyID, antigenID, heavyChainDistanceToAntigen)
+
+            paddedArrays = paddingOfArray(lightChainDistanceToAntigen,heavyChainDistanceToAntigen)
+
+            structures.append([paddedArrays[0],paddedArrays[1]])
+            CDRcounter = CDRcounter +1
+
+
+        longestAntigen = 0
+        longestChain = 0
+
+        for x in structures:
+            if len(x[0]) > longestAntigen:
+                longestAntigen = len(x[0])
+            if len(x[0][0]) > longestChain:
+                longestChain = len(x[0][0])
+
+        paddedStructures = []
+        for x in structures:
+            paddedStructures.append(expandAntibodyChainToMaxLength(x, longestAntigen, longestChain))
+
+
+
+        numpArray = np.array(paddedStructures)
+
+        encoded_dataset = PDBDataset(self, dataset.pdbFilePaths,dataset.labels,dataset.metadata_file,EncodedData(numpArray,None,dataset.get_example_ids()))
+
+        #print("Shape of encoded_dataset.encoded_data.examples: " , numpy.shape(encoded_dataset.encoded_data.examples))
+        return encoded_dataset
+
+
+def caluculateDistance(collection, chain, antigen, list):
+    for x in range(len(collection[antigen])):
+        for i in range(len(collection[chain])):
+            a = np.array((collection[chain][i][0], collection[chain][i][1], collection[chain][i][2]))
+            b = np.array((collection[antigen][x][0], collection[antigen][x][1], collection[antigen][x][2]))
+
+            rounded = round(np.linalg.norm(a - b), 2)
+            list[x][i] = rounded
+
+
+"""
+This method takes in two 2D arrays as input. The format should be the light chain first and heavy second
+Then pads the shortest one to the length of the longest one
+It returns a tuple like (lightChain,HeavyChain)
+
+The arrays should be numpy arrays.
+The arrays should have the same amount of rows, but not columns
+It is the individual rows that gets padded values to match the longer one
+
+"""
+def paddingOfArray(lightChain, heavyChain):
+    diff = len(lightChain[0]) - len(heavyChain[0])
+
+    if diff > 0:
+        longest = lightChain
+        shortest = heavyChain
+        paddedShortest = []
```

Comment:
repeated code is dangerous and should be avoided at all cost. you can ask Geir Kjetil about this one time he was 100% sure he had fixed a certain bug in his code, only to find out after a very long time that the bug was not fixed in the copied code. 
it can be solved very easily though by putting the repeated code in a function and calling it twice.